### PR TITLE
feat: implemented Pizza GitHub action

### LINKED ISSUE #114: Feature: GitHub action for `pizza`
### Suggested solution

Stub

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] 📦 Chore
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes https://github.com/open-sauced/pizza-cli/issues/114

## Steps to QA

Right now I'm testing on a personal repo.

## Added to documentation?

- [x] 📜 README.md
- [ ] 📜 CONTRIBUTING.md
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by jpmcb — APPROVED** [review]

File: index.js
Comment (nickytonline) [inline index.js]: I've tried:

```javascript
await exec.exec("pizza", args);
```

and

```javascript
await exec.exec(`pizza ${pizzaArgs}`);
```
and the equivalents with `execSync` etc. but they all result in running the pizza CLI without the args and the args are present in `pizzaArgs`. I'm sure it's something super silly too @brandonroberts @jpmcb.

```
Run node /home/runner/work/_actions/open-sauced/pizza-action/nickytonline/implement-action/index.js
Running Pizza CLI with args: [ 'generate', 'codeowners', './' ]
/opt/hostedtoolcache/node/20.17.0/x64/bin/pizza generate codeowners ./
A command line utility for insights, metrics, and all things OpenSauced
Usage:
  pizza <command> <subcommand> [flags]
Available Commands:
  bake        Use a pizza-oven to source git commits into OpenSauced
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command
  insights    Gather insights about git contributors, repositories, users and pull requests
  login       Log into the CLI application via GitHub
  repo-query  Ask questions about a GitHub repository
  show        Get visual metrics of a repository
  version     Displays the build version of the CLI
Flags:
      --beta                Shorthand for using the beta OpenSauced API
                            endpoint
                            ("https://beta.api.opensauced.pizza/").
                            Supersedes the '--endpoint' flag
      --disable-telemetry   Disable sending telemetry data to OpenSauced
  -e, --endpoint string     The API endpoint to send requests to (default
                            "https://api.opensauced.pizza/")
  -h, --help                help for pizza
  -o, --output string       The formatting for command output. One of:
                            (table, yaml, csv, json) (default "table")
Use "pizza [command] --help" for more information about a command.
/usr/bin/git config --global user.name github-actions
/usr/bin/git config --global user.email github-actions@github.com
/usr/bin/git add .
/usr/bin/git commit -m Auto-commit from OpenSauced Pizza Action
On branch main
Your branch is up to date with 'origin/main'.
nothing to commit, working tree clean
Error: The process '/usr/bin/git' failed with exit code 1
Error: Process completed with exit code 1.
```

It seems to run fine too if I run it with Node locally. Maybe it's an encoding issue. I'll try explicitly running it as UTF-8.

Comment (nickytonline) [reply index.js]: I'll be deleting this test repo once this is merged, but you can see the different tries. 😅 https://github.com/nickytonline/test-action/actions/workflows/pizza.yml

Going to keep digging for the time being.

Comment (nickytonline) [reply index.js]: I'm going to try running it from a bash script.

Comment (nickytonline) [reply index.js]: Being explicit about the encoding changes nothing for the args not being passed in. Going to try the bash script on the fly option.

Comment (nickytonline) [reply index.js]: I think I figured it out while trying the bash way. I need to strip the quotes.

```
pizza "generate" "codeowners" "./"
Script path: /home/runner/work/test-action/test-action/run_pizza.sh
/usr/bin/bash /home/runner/work/test-action/test-action/run_pizza.sh
A command line utility for insights, metrics, and all things OpenSauced
Usage:
  pizza <command> <subcommand> [flags]
Available Commands:
  bake        Use a pizza-oven to source git commits into OpenSauced
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command
  insights    Gather insights about git contributors, repositories, users and pull requests
  login       Log into the CLI application via GitHub
  repo-query  Ask questions about a GitHub repository
  show        Get visual metrics of a repository
  version     Displays the build version of the CLI
Flags:
      --beta                Shorthand for using the beta OpenSauced API
                            endpoint
                            ("https://beta.api.opensauced.pizza/").
                            Supersedes the '--endpoint' flag
      --disable-telemetry   Disable sending telemetry data to OpenSauced
  -e, --endpoint string     The API endpoint to send requests to (default
                            "https://api.opensauced.pizza/")
  -h, --help                help for pizza
  -o, --output string       The formatting for command output. One of:
                            (table, yaml, csv, json) (default "table")
```

Comment (nickytonline) [reply index.js]: Hmm, still no dice.

```
Raw pizza-args: generate codeowners ./
Stripped args: [ 'generate', 'codeowners', './' ]
Bash script content: #!/bin/bash
set -e
pizza generate codeowners ./
Executing bash script...
/usr/bin/bash /home/runner/work/test-action/test-action/run_pizza.sh
A command line utility for insights, metrics, and all things OpenSauced
Usage:
  pizza <command> <subcommand> [flags]
Available Commands:
  bake        Use a pizza-oven to source git commits into OpenSauced
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command
  insights    Gather insights about git contributors, repositories, users and pull requests
  login       Log into the CLI application via GitHub
  repo-query  Ask questions about a GitHub repository
  show        Get visual metrics of a repository
  version     Displays the build version of the CLI
Flags:
      --beta                Shorthand for using the beta OpenSauced API
                            endpoint
                            ("https://beta.api.opensauced.pizza/").
                            Supersedes the '--endpoint' flag
      --disable-telemetry   Disable sending telemetry data to OpenSauced
  -e, --endpoint string     The API endpoint to send requests to (default
                            "https://api.opensauced.pizza/")
  -h, --help                help for pizza
  -o, --output string       The formatting for command output. One of:
                            (table, yaml, csv, json) (default "table")
Use "pizza [command] --help" for more information about a command.
/usr/bin/git config --global user.name github-actions
/usr/bin/git config --global user.email github-actions@github.com
/usr/bin/git add .
/usr/bin/git commit -m Auto-commit from OpenSauced Pizza Action
```

This looks good though 🤔 

```
Raw pizza-args: generate codeowners ./
Stripped args: [ 'generate', 'codeowners', './' ]
Bash script content: #!/bin/bash
set -e
pizza generate codeowners ./
Executing bash script...
/usr/bin/bash /home/runner/work/test-action/test-action/run_pizza.sh
```

I'll get back on this tomorrow morning.

**Review by brandonroberts — APPROVED** [review]

File: index.js
Comment (jpmcb) [inline index.js]: Thinking out loud: I'm wondering if instead of making committing the generated codeowners a required part of the flow for this action, we can instruct users how they can commit the changes with a followup action like:

```yaml
      - name: Commit CODEOWERS
        run: |
          git config --global user.name github-actions
          git config --global user.email github-actions@github.com
          git add .
          git commit -m "Auto-commit OpenSauced generated CODEOWNERS"
          git push
```

I'd imagine there's alot of different things people will want to do with a generated codeowners file: alert based on changes, push it to some "reports" bucket in the cloud, etc. etc.

And different people / orgs will have wildly different permissions and capabilities in their GitHub actions. We _probably_ don't want to be in the business of fleshing out every possible git config and GitHub auth scenario here.

Thoughts?

Comment (nickytonline) [reply index.js]: Yeah, that sounds good. Also. the above was going to become a bash script instead. I was working on wrangling the install of the CLI, but that whole index.js doesn't need to exist.

Comment (nickytonline) [reply index.js]: It can be a script that ships with the action and they can use it if they want, or do there own thing like you mentioned.

**Comment by nickytonline** [comment]
Blocked by https://github.com/open-sauced/pizza-cli/pull/126. Should be merged and released soon.

File: action.yml
Comment (nickytonline) [inline action.yml:46]: We can commit the codeowners or future actions the pizza CLI does for a repo, but by default it's set to false.

File: action.yml
Comment (nickytonline) [inline action.yml:49]: This is minimal boilerplate for committing and pushing so I kept it inline the action.yml. It's also an optional step.

File: README.md
Comment (nickytonline) [inline README.md:30]: This is the default value so no need to include it.

File: README.md
Comment (nickytonline) [inline README.md:30]: Put in the example to commit and push.

**Comment by nickytonline** [comment]
Things run successfully now.

CleanShot 2024-08-30 at 15 43 54

File: action.yml
Comment (nickytonline) [inline action.yml:35]: I left this as 	pizza-cli` for now @jpmcb because `go install` doesn't support it and I wanted to install a version instead of build it where I could use `-o`.

File: README.md
Comment (nickytonline) [inline README.md:23]: copy pasta error. I started with my other action as a template.

**Pull request merged**
